src/Chapter2.hs

```diff
@@ -621,7 +641,15 @@ Write a function that takes elements of a list only in even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [a] -> [a]
+takeEven [] =[]
+takeEven (x:xs) = recursiveEven 0 (x : xs)
+    where
+        recursiveEven :: Int -> [a] -> [a]
+        recursiveEven _ [] = []
+        recursiveEven c (x:xs) = if mod c 2 == 0
+            then x : recursiveEven (c+1) xs
+            else recursiveEven (c+1) xs
```

Comment:
Nice idea on the helper functions :+1:

A few additional thoughts on your idea: as you have only two states of the operation (odd or even), instead of accumulators of the current position number, you can just use `Bool`. And on each next recursion call, you just give it `not acc`. This will remove the necessity to check `mod 2` each time, which would save a couple of functions call on each interaction.

Alternatively, you could use pattern matching in `takeEven` instead. As you need to take every second element, it is possible to represent with a few pattern matching clauses 🙂

Reply:
>> Alternatively, you could use pattern matching in takeEven instead. 
>> As you need to take every second element, it is possible to represent with a few pattern matching clauses


I tried to use the pattern but I didn't understand

takeEven :: [a] -> [a]
takeEven [] = []
takeEven (x:y:xs) = [y] ++ takeEven xs

ghci> takeEven [2, 1, 3, 5, 6]
[1,5*** Exception: src\Chapter2.hs:(654,1)-(655,38): Non-exhaustive patterns in function takeEven

Reply:
I got it!! A clean and readeble solution. Thx for support